src/state/assets/userAssets.ts

```diff
@@ -121,227 +122,257 @@ function deserializeUserAssetsState(serializedState: string) {
 
-        const isMap = userAssets instanceof Map;
-        const allIdsArray = isMap ? Array.from(userAssets.keys()) : userAssets.map(asset => asset.uniqueId);
-        const userAssetsMap = isMap ? userAssets : new Map(userAssets.map(asset => [asset.uniqueId, asset]));
+          idsByChain.set('all', allIdsArray);
 
-        idsByChain.set('all', allIdsArray);
+          const smallBalanceThreshold = supportedNativeCurrencies[reduxStore.getState().settings.nativeCurrency].userAssetsSmallThreshold;
 
-        const smallBalanceThreshold = supportedNativeCurrencies[store.getState().settings.nativeCurrency].userAssetsSmallThreshold;
+          const filteredAllIdsArray = allIdsArray.filter(id => {
+            const asset = userAssetsMap.get(id);
+            return asset && (+asset.native?.balance?.amount ?? 0) > smallBalanceThreshold;
+          });
 
-        const filteredAllIdsArray = allIdsArray.filter(id => {
-          const asset = userAssetsMap.get(id);
-          return asset && (+asset.native?.balance?.amount ?? 0) > smallBalanceThreshold;
-        });
+          const searchCache = new Map<string, UniqueId[]>();
 
-        const searchCache = new Map<string, UniqueId[]>();
+          Array.from(chainBalances.keys()).forEach(userAssetFilter => {
+            const filteredIds = (idsByChain.get(userAssetFilter) || []).filter(id => filteredAllIdsArray.includes(id));
+            searchCache.set(`${userAssetFilter}`, filteredIds);
+          });
 
-        Array.from(chainBalances.keys()).forEach(userAssetFilter => {
-          const filteredIds = (idsByChain.get(userAssetFilter) || []).filter(id => filteredAllIdsArray.includes(id));
-          searchCache.set(`${userAssetFilter}`, filteredIds);
-        });
+          searchCache.set('all', filteredAllIdsArray);
 
-        searchCache.set('all', filteredAllIdsArray);
-
-        if (isMap) {
-          return { associatedWalletAddress, chainBalances, idsByChain, searchCache, userAssets };
-        } else
-          return {
-            associatedWalletAddress,
-            chainBalances,
-            idsByChain,
-            searchCache,
-            userAssets: userAssetsMap,
-          };
-      }),
-  }),
-  {
-    deserializer: deserializeUserAssetsState,
-    partialize: state => ({
-      associatedWalletAddress: state.associatedWalletAddress,
-      chainBalances: state.chainBalances,
-      idsByChain: state.idsByChain,
-      userAssets: state.userAssets,
+          if (isMap) {
+            return { chainBalances, idsByChain, searchCache, userAssets };
+          } else
+            return {
+              chainBalances,
+              idsByChain,
+              searchCache,
+              userAssets: userAssetsMap,
+            };
+        }),
     }),
-    serializer: serializeUserAssetsState,
-    storageKey: 'userAssets',
-    version: 3,
+    {
+      storageKey: `userAssets_${address}`,
+      version: 0,
```

Comment:
might need a version bump in the persister here

Reply:
correct me if i'm wrong but i think 0 is correct since these storageKeys haven't been used before?

Reply:
oh true i didn't see you changed the storageKey. Disregard